tests/shared/models/selection-state.spec.js

```diff
@@ -370,6 +370,42 @@ describe('Model: selectionState', function () {
         m.chai.expect(imagePath).to.equal('foo.zip')
       })
 
+      it('should infer a compressed raw image if the penultimate extension is missing', function () {
```

Comment:
I'm confused - what part of this unit-test is checking that it's inferring a raw image type?

Reply:
It's in `selectionState.setImage()` – there are a bunch of checks in `lib/shared/store.js` (just below the TODO comment I added there) that'd previously go off with "invalid archive extension", which makes the tests very opaque here.

Reply:
> which makes the tests very opaque here.

Ahh, fair enough :)